apps/catalog/catalog/src/app/marketplace/offer/shell/contract-edit/contract-edit.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Component, ChangeDetectionStrategy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { OfferShellComponent } from '../shell.component';
+import { filter, first, map, pluck } from 'rxjs/operators';
+import { NegotiationGuardedComponent } from '@blockframes/contract/negotiation/guard';
+import { NegotiationForm } from '@blockframes/contract/negotiation';
+import { MatDialog } from '@angular/material/dialog';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { ConfirmDeclineComponent } from '@blockframes/contract/contract/components/confirm-decline/confirm-decline.component';
+import { NegotiationService } from '@blockframes/contract/negotiation/+state/negotiation.service';
+import { OrganizationQuery } from '@blockframes/organization/+state';
+import { ConfirmComponent } from '@blockframes/ui/confirm/confirm.component';
+import { combineLatest } from 'rxjs';
+
+@Component({
+  selector: 'catalog-contract-edit',
+  templateUrl: './contract-edit.component.html',
+  styleUrls: ['./contract-edit.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ContractEditComponent implements NegotiationGuardedComponent, OnInit {
+
+  activeOrgId = this.query.getActiveId();
+  form = new NegotiationForm();
+  contracts$ = this.shell.offer$.pipe(
+    map(offer => offer.contracts),
+  );
+  contractId$ = this.route.params.pipe(pluck('saleId'));
+  sale$ = combineLatest([
+    this.contracts$,
+    this.contractId$
+  ]).pipe(
+    map(([contracts, id]) => contracts.find(contract => contract.id === id))
+  );
+  negotiation$ = this.sale$.pipe(map(contract => contract.negotiation));
+
+  constructor(
+    private snackBar: MatSnackBar,
+    private shell: OfferShellComponent,
+    private negotiationService: NegotiationService,
+    private query: OrganizationQuery,
+    private dialog: MatDialog,
+    private router: Router,
+    private route: ActivatedRoute,
+  ) { }
+
+  async ngOnInit(): Promise<void> {
+    const negotiation = await this.negotiation$.pipe(
+      filter(data => !!data),
+      first()
+    ).toPromise();
+    this.form.hardReset(negotiation);
+  }
+
+  async decline() {
+    this.form.markAsPristine(); // usefull to be able to route in the NegotiationGuard
+    const sale = await this.sale$.pipe(first()).toPromise();
+    const ref = this.dialog.open(ConfirmDeclineComponent);
+    const options = { params: { contractId: sale.id } };
+    ref.afterClosed().subscribe(declineReason => {
+      if (typeof declineReason === 'string') {
+        this.negotiationService.update(
+          sale.negotiation.id, { declineReason, status: 'declined' }, options
+        );
```

Comment:
Formatting is difficult to read. Create intermediary variables to make it easier to read.
```suggestion
        const id = sale.negotiation.id;
        this.negotiationService.update(id, { declineReason, status: 'declined' }, options);
```